drivers/ethernet/eth_xlnx_gem_priv.h

```diff
@@ -511,11 +518,28 @@ struct eth_xlnx_dma_area_gem##port {\
 		& ~(ETH_XLNX_BUFFER_ALIGNMENT - 1))];\
 };
 
-/* DMA memory area instantiation macro */
+#if defined(CONFIG_SOC_XILINX_ZYNQMP)
+#define ETH_XLNX_GEM_DMA_AREA_Q1_DECL(port) \
+struct eth_xlnx_dma_q1_area_gem##port {\
+	struct eth_xlnx_gem_bd tx_bd[1];\
+};
+#else
+#define ETH_XLNX_GEM_DMA_AREA_Q1_DECL(port);
+#endif
+
+/* DMA memory area instantiation macros */
 #define ETH_XLNX_GEM_DMA_AREA_INST(port) \
 static struct eth_xlnx_dma_area_gem##port eth_xlnx_gem##port##_dma_area\
 	__ocm_bss_section __aligned(4096);
 
+#if defined(CONFIG_SOC_XILINX_ZYNQMP)
+#define ETH_XLNX_GEM_DMA_AREA_Q1_INST(port) \
+static struct eth_xlnx_dma_q1_area_gem##port eth_xlnx_gem##port##_q1_dma_area\
+	__ocm_bss_section __aligned(8);
```

Comment:
This attribute has turned out as a mistake after another look at the Zynq's Technical Reference Manual.

The entire DMA area of each GEM is always 4k-aligned so that the properties of the respective MMU page table entry can be modified without affecting any non-related data (at least in the Zynq, whose A9 cores have a MMU). With the simplest RX/TX buffer descriptor format which is currently the only one supported by the driver, each descriptor is 8 bytes in size, so the top address of the descriptor rings, which also equals the bottom address of the RX packet data buffers which are next in the DMA area struct, is always 8-byte aligned.

Even if an odd number is specified in the device tree for the rx-buffer-size and tx-buffer-size properties, each RX/TX packet data buffer will always be padded to a 4 byte boundary (that's what the ETH_XLNX_BUFFER_ALIGNMENT define is for). So the top address of the TX packet data buffers, which are last in the DMA area struct, is always at least 4-byte aligned.

The modified ETH_XLNX_GEM_INITIALIZE macro now first places the 4k-aligned, actually useful DMA area in the OCM memory, directly followed by the dummy DMA area which contains only the one TX descriptor. The __aligned(8) attribute should have enforced placement of the dummy DMA area at an 8 byte-aligned address in case the top address of the actual DMA area was only 4 byte-aligned. I did believe that the TRM stated that the base address of each buffer descriptor ring must be 8-byte aligned, however, the documentation of the rx_qbar and tx_qbar registers in the Zynq's TRM states "The descriptors should be aligned at 32-bit boundaries". Therefore, this attribute is obsolete, I'll remove it.